schematic/scheme/gschem/builtins.scm

```diff
@@ -310,8 +315,51 @@
 (define-action-public (&attributes-attach #:label (_ "Attach Attributes") #:icon "attribute-attach")
   (%attributes-attach))
 
-(define-action-public (&attributes-detach #:label (_ "Detach Attributes") #:icon "attribute-detach")
-  (%attributes-detach))
+
+
+( define-action-public
+  (
+    &attributes-detach
+    #:label (_ "Detach Attributes")
+    #:icon "attribute-detach"
+  )
+
+  ( let*
+    (
+    ( page ( active-page ) )
+    ( sel  ( if page (page-selection page) '() ) )
+    ( attrs '() )
+    )
+
+    ( define ( is-attr-to-detach obj ) ; predicate
+      ; return:
+      ( and
+        ( attribute?        obj ) ; if it's attribute
+        ( text-visible?     obj ) ; if it's visible
+        ( attrib-attachment obj ) ; and attached to some object
+      )
+    )
+
+    ( define ( detach-attr attr )
+      ( detach-attribs! (attrib-attachment attr) attr )
+      ( log! 'message (_ "Attribute detached: [~a]") (text-string attr) )
+      ( deselect-object! attr )
+    )
+
+
+    ( set! attrs (filter is-attr-to-detach sel) )
+
+    ( when ( > (length attrs) 0 )
```

Comment:
This can be safely replaced with `(unless (null? attrs) ...)`, which seems to be a bit more readable.